syntax = "proto3";

message RootMessage {
  oneof root_field {
    int32 int_value = 1;
    string string_value = 2;
  }

  message NestedMessage {
    oneof nested_field {
      float float_value = 1;
      bool bool_value = 2;
    }
  }

  NestedMessage nested_message = 3;
  string optional_field = 4;
  int64 non_optional_field = 5;
}
